src/ac_copilot_trainer/modules/realtime_coaching.lua

```
@@ -1,360 +1,344 @@
-local buckets = {}                 ---@type (integer|nil)[]
-local indexedSegments = {}         ---@type table[]
-local indexBuilt = false
-
--- Precomputed sorted list of brake/corner segment spline starts for O(1) approach scan.
-local brakeCornerStarts = {}       ---@type {s0: number, idx: integer}[]
-
--- Precomputed brake segment index -> corner label map for O(1) label lookup.
-local brakeToCornerLabel = {}      ---@type table<integer, string>
+local lastView = nil               -- last returned viewmodel (for dedupe / fade)
+local lastEmittedKey = nil         -- {kind .. ":" .. cornerLabel}
+local lastEmittedAt = -1e9         -- monotonic time of last emission (seconds)
```

Luacheck (1.2.0)

[warning] unused variable 'lastEmittedAt'

(W211)